src/fee.rs

```diff
@@ -18,6 +19,14 @@ pub fn calculate_pool_trade_fee(amount: Balance, fee: (u32, u32)) -> Option<Bala
         .checked_mul(numerator as Balance)
 }
 
+pub fn multiply_rational(numerator: u128, denominator: u128, amount: Balance) -> Option<Balance> {
+    let (numerator, denominator, amount) = to_u256!(numerator, denominator, amount);
+
+    let result = amount.checked_mul(numerator).and_then(|v| v.checked_div(denominator))?;
+
+    to_balance!(result).ok()
+}
```

Comment:
it's very similar to [`multiply_by_rational_with_rounding`](https://paritytech.github.io/substrate/master/sp_arithmetic/helpers_128bit/fn.multiply_by_rational_with_rounding.html), do we want a new function for that?

Reply:
i would try to avoid if possible.  I would prefer to consolidate and have one way.

Reply:
Just to make it clear, do we want to implement it for `Ratio` or use `multiply_by_rational_with_rounding` instead? I suggest implement a new method for `Ratio` and use `multiply_by_rational_with_rounding` in the implementation of this method.

Reply:
sure we can implement it on `Ratio`
(would do it in a new PR, though)